app/types/remix-routes.d.ts

```diff
@@ -0,0 +1,241 @@
+declare module "remix-routes" {
+  type URLSearchParamsInit = string | string[][] | Record<string, string> | URLSearchParams;
+  // symbol won't be a key of SearchParams
+  type IsSearchParams<T> = symbol extends keyof T ? false : true;
+  
+  type ExportedQuery<T> = IsSearchParams<T> extends true ? T : never;
+  
+
+  export interface Routes {
+  
+    "/": {
+      params: never,
+      query: ExportedQuery<import('../root').SearchParams>,
+    };
+  
+    "/dashboard": {
+      params: never,
+      query: ExportedQuery<import('../routes/dashboard/index').SearchParams>,
+    };
+  
+    "/exchange": {
+      params: never,
+      query: ExportedQuery<import('../routes/exchange/index').SearchParams>,
+    };
+  
+    "/exchange/select": {
+      params: never,
+      query: ExportedQuery<import('../routes/exchange/select').SearchParams>,
+    };
+  
+    "/exchange/widget/:chainId/:assetId": {
+      params: {
+        chainId: string | number;assetId: string | number;
+      } ,
+      query: ExportedQuery<import('../routes/exchange/widget.$chainId.$assetId').SearchParams>,
+    };
+  
+    "/gifts": {
+      params: never,
+      query: ExportedQuery<import('../routes/gifts/index').SearchParams>,
+    };
+  
+    "/gifts/details": {
+      params: never,
+      query: ExportedQuery<import('../routes/gifts/details').SearchParams>,
+    };
+  
+    "/onboarding": {
+      params: never,
+      query: ExportedQuery<import('../routes/onboarding/index').SearchParams>,
+    };
+  
+    "/onboarding/create-wallet": {
+      params: never,
+      query: ExportedQuery<import('../routes/onboarding/create-wallet').SearchParams>,
+    };
+  
+    "/onboarding/password": {
+      params: never,
+      query: ExportedQuery<import('../routes/onboarding/password').SearchParams>,
+    };
+  
+    "/onboarding/restore/:mnemonic": {
```

Comment:
Is it safe to pass sensitive data in the path? Can it be the case that UI might show the path somewhere on the screen and thus compromising the mnemonic

Reply:
It shouldn't be visible easily since it's an iframe, but if we go to a network tab and see the request, you will see a mnemonic. So, still kinda unsafe way

Reply:
Removed url param, pass mnemonic through `history API` ([doc](https://developer.mozilla.org/en-US/docs/Web/API/History)). This way nothing is exposed to the user.